x/slashing/abci.go

```diff
@@ -23,3 +23,27 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 		k.HandleValidatorSignature(ctx, voteInfo.Validator.Address, voteInfo.Validator.Power, voteInfo.SignedLastBlock)
 	}
 }
+
+// Called every block, update validator set
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) []abci.ValidatorUpdate {
+	if ctx.BlockHeight()%10 == 0 { // TODO should update hardcoded 10 to stakingParams.EpochInterval (epoch_interval)
+		// execute all epoch actions
+		iterator := k.GetEpochActionsIterator()
+	
+		for ; iterator.Valid(); iterator.Next() {
+			var msg sdk.Msg
+			bz := iterator.Value()
+			k.cdc.MustUnmarshalBinaryBare(bz, &msg)
+
+			switch msg := msg.(type) {
+			case *types.MsgUnjail:
+				res, err := k.EpochUnjail(sdk.WrapSDKContext(ctx), msg)
```

Comment:
Don't need res, and should catch err

Reply:
How should we report the error to the validator ?